src/v/storage/tests/index_state_test.cc

```diff
@@ -17,6 +20,21 @@
 
 #include <boost/test/unit_test.hpp>
 
+#ifdef NDEBUG
+inline constexpr int num_test_runs = 100;
+inline constexpr int num_samples = 10000;
+#else
+inline constexpr int num_test_runs = 1;
```

Comment:
i'd prefer in debug to run at least once with each index type

debug has more assertions enabled, sanitizers, etc.

something which should work although i don't like is having use_compression be a static int which is incremented % 2. no need for randomness